src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+        println("# ]")
+        #
+        # Write tables
+        #  Reference value and precision match tables are separate since it is more
+        #  common to update reference values occiasionally while precision values are
+        #  typically changed rarely and the precision values are hand edited from experience.
+        #
+        # Write table of reference values
+        println("varr = [")
+        for lv in cb.func.curStats_flat
+            s1 = lv[1]
+            l1 = length(s1)
+            s1 = sp[1:(a1l - l1)] * "\"" * s1 * "\""
+            s2 = lv[2]
+            if typeof(s2) == String
+                l2 = length(s2)
+                s2 = sp[1:(a2l - l2)] * "\"" * s2 * "\""
+                s22 = ""
+            end
+            if typeof(s2) == Symbol
+                s22 = s2
+                l2 = length(String(s2))
+                s2 = sp[1:(a2l - l2 + 1)] * ":"
+            end
+            s3 = sprintf1(fmt3, lv[3])
+            s4 = sprintf1(fmt3, lv[4])
+            s5 = sprintf1(fmt3, lv[5])
+            s6 = sprintf1(fmt3, lv[6])
+            println(
+                " [ ",
+                s1,
+                ", ",
+                s2,
+                s22,
+                ",",
+                s3,
+                ",",
+                s4,
+                ",",
+                s5,
+                ",",
+                s6,
+                " ],",
+            )
+        end
+        println("]")
+
+        # Write table of reference match precisions using default precision that
+        # can be hand upadated.
+        println("parr = [")
+        for lv in cb.func.curStats_flat
+            s1 = lv[1]
+            l1 = length(s1)
+            s1 = sp[1:(a1l - l1)] * "\"" * s1 * "\""
+            s2 = lv[2]
+            if typeof(s2) == String
+                l2 = length(s2)
+                s2 = sp[1:(a2l - l2)] * "\"" * s2 * "\""
+                s22 = ""
+            end
+            if typeof(s2) == Symbol
+                s22 = s2
+                l2 = length(String(s2))
+                s2 = sp[1:(a2l - l2 + 1)] * ":"
+            end
+            s3 = sprintf1(fmt3, lv[3])
+            s4 = sprintf1(fmt3, lv[4])
+            s5 = sprintf1(fmt3, lv[5])
+            s6 = sprintf1(fmt3, lv[6])
+            println(
+                " [ ",
+                s1,
+                ", ",
+                s2,
+                s22,
+                ",",
+                "    16,    16,    16,    16 ],",
+            )
+        end
+        println("]")
+        println("# END SCPRINT")
+    end
+end
+
+"""
+ scdocheck :: Compare a current State check call-back set of values with a 
+              reference set and match precision table pair.
+               
+              Input:  cb - StateCheck call-back variables
+                      refDat - table of reference values and match precisions
+
+              Output: true  - comparison passed
```

Comment:
Can we include what the metric that results in a pass/fail comparison is? It might be good to include an example here